### PR TITLE
Bump go modules

### PR DESCRIPTION
**Pull request by howardjohn**
Mostly changes to gateway-api removing ReferencePolicy

Blocked by https://github.com/kubernetes-sigs/gateway-api/pull/1472, we need to pull that in

### CODE REVIEW

**Review by jacob-delgado — APPROVED** [review]
The blocking PR got merged in so you should be close to being able to push this in too.

File: licenses/istio.io/client-go/LICENSE [hunk 1]
```diff
@@ -187,7 +187,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2016-2022 Istio Authors
+   Copyright 2016-2020 Istio Authors
```
Comment (lei-tang) [inline licenses/istio.io/client-go/LICENSE:190]: Is changing 2022 to 2020 needed?

Comment (howardjohn) [reply licenses/istio.io/client-go/LICENSE:190]: Seems odd but it's auto generated. Needs to be fixed in client-go?

Comment (howardjohn) [reply licenses/istio.io/client-go/LICENSE:190]: Oops - I downgraded client-go accidentally. I'll fix that. Nice catch

File: pkg/wasm/imagefetcher.go [hunk 2]
```diff
@@ -73,7 +74,7 @@ func NewImageFetcher(ctx context.Context, opt ImageFetcherOption) *ImageFetcher
 	}
 
 	if opt.Insecure {
-		t := remote.DefaultTransport.Clone()
+		t := remote.DefaultTransport.(*http.Transport).Clone()
```
Comment (hzxuzhonghu) [inline pkg/wasm/imagefetcher.go:77]: redundant cast

Comment (howardjohn) [reply pkg/wasm/imagefetcher.go:77]: It's not.. the changes the type to an interface that doesn't have Clone but the underlying type has Clone.  

Their own library uses the exact same code as here

**Review by ericvn — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]

**Pull request merged**